ggshield/cmd/secret/scan/docset.py

```diff
@@ -0,0 +1,86 @@
+import json
+from typing import Callable, Iterable, Iterator, List
+
+import click
+
+from ggshield.core.constants import MAX_WORKERS
+from ggshield.core.text_utils import create_progress_bar
+from ggshield.core.utils import ScanContext, ScanMode, handle_exception
+from ggshield.output import OutputHandler
+from ggshield.scan import File, ScanCollection
+from ggshield.scan.scannable import Scanner
+
+
+def generate_files_from_docsets(path: str, verbose: bool = False) -> Iterator[File]:
+    with open(path, "r") as file:
+        for line in file:
+            obj = json.loads(line)
+            documents = obj["documents"]
+            for document in documents:
+                if verbose:
+                    click.echo(f"  * {document['id']}", err=True)
+                yield File(document["content"], document["id"])
+
+
+def create_scans_from_docset_paths(
+    scanner: Scanner,
+    paths: Iterable[str],
+    progress_callback: Callable[[], None],
+    verbose: bool = False,
+) -> List[ScanCollection]:
+    scans: List[ScanCollection] = []
+
+    for path in paths:
+        if verbose:
+            click.echo(f"- {click.format_filename(path)}", err=True)
+
+        files = generate_files_from_docsets(path, verbose)
+        results = scanner.scan(
+            files, scan_threads=MAX_WORKERS, progress_callback=lambda **x: None
+        )
+        scans.append(ScanCollection(id=path, type="docset", results=results))
+        progress_callback()
+
+    return scans
+
+
+@click.command()
+@click.argument(
+    "paths", nargs=-1, type=click.Path(exists=True, resolve_path=True), required=True
+)
+@click.pass_context
+def docset_cmd(ctx: click.Context, paths: List[str]) -> int:  # pragma: no cover
```

Comment:
Sorry to jump in the discussion. When testing locally with a jsonl input, the progress bar does not behave correctly :(   

Also, and it is linked, I am wondering why we are using the `Scanner.scan` interface instead of the `Files.scan` interface as in other commands implementation (see pypi for instance).

Reply:
Regarding `Scanner.scan()`, I think it's OK: when #344 is done, `Files.scan()` will be gone and all commands should do what this one does.

Reply:
The progress bar should be fixed now. It advances per input file though, not per line (as computing the total number of line before hand would be expensive).